velox/dwio/parquet/reader/PageReader.cpp

```diff
@@ -251,10 +251,7 @@ void PageReader::prepareDataPageV1(const PageHeader& pageHeader, int64_t row) {
   if (!hasChunkRepDefs_ && (numRowsInPage_ == kRowsUnknown || maxDefine_ > 1)) {
     readPageDefLevels();
   }
-
-  if (row != kRepDefOnly) {
-    makeDecoder();
-  }
+  makeDecoder();
```

Comment:
I remember we do need to skip some work when it is `kRepDefOnly` to keep it efficient.  Can you try to refine the condition so that the decoders are constructed when it is not `kRepDefOnly` or it is a IS NULL call?

Reply:
Added a check for 'toSkip', and if it is 0, the skip will not be called. Do you think it makes sense? Thanks.